Presentation.Web/Controllers/API/V1/OrganizationRegistrationController.cs

```diff
@@ -0,0 +1,137 @@
+﻿using System;
+using System.Linq;
+using System.Net;
+using System.Net.Http;
+using System.Web.Http;
+using Core.ApplicationServices.Model.Organizations;
+using Core.ApplicationServices.Organizations;
+using Core.DomainModel.ItContract;
+using Core.DomainModel.Organization;
+using Core.DomainServices;
+using Presentation.Web.Controllers.API.V1.Mapping;
+using Presentation.Web.Infrastructure.Attributes;
+using Presentation.Web.Models.API.V1;
+using Presentation.Web.Models.API.V1.Organizations;
+using Swashbuckle.Swagger.Annotations;
+
+namespace Presentation.Web.Controllers.API.V1
+{
+    [PublicApi]
+    [RoutePrefix("api/v1/organizations/{organizationUuid}/organization-units/{unitUuid}")]
+
+    public class OrganizationRegistrationController: BaseApiController
+    {
+        private readonly IOrganizationUnitService _organizationUnitService;
+        private readonly IOrgUnitService _orgUnitService;
+
+        public OrganizationRegistrationController(IOrganizationUnitService organizationUnitService,
+            IOrgUnitService orgUnitService)
+        {
+            _organizationUnitService = organizationUnitService;
+            _orgUnitService = orgUnitService;
+        }
+
+        [HttpGet]
+        [Route("registrations")]
+        [SwaggerResponse(HttpStatusCode.OK)]
+        [SwaggerResponse(HttpStatusCode.Forbidden)]
+        [SwaggerResponse(HttpStatusCode.NotFound)]
+        public HttpResponseMessage GetRegistrations(Guid organizationUuid, Guid  unitUuid)
+        {
+            return _organizationUnitService.GetRegistrations(organizationUuid, unitUuid)
+                .Select(ToRegistrationDto)
+                .Match(Ok, FromOperationError);
+        }
+
+        [HttpDelete]
+        [Route("registrations")]
+        [SwaggerResponse(HttpStatusCode.OK)]
+        [SwaggerResponse(HttpStatusCode.Forbidden)]
+        [SwaggerResponse(HttpStatusCode.NotFound)]
+        public HttpResponseMessage RemoveRegistrations(Guid organizationUuid, Guid unitUuid, [FromBody] ChangeOrganizationRegistrationRequestDTO requestDto)
+        {
+            var changeParameters = ToChangeParameters(requestDto);
+            return _organizationUnitService.DeleteRegistrations(organizationUuid, unitUuid, changeParameters)
+                .Match(FromOperationError, Ok);
+        }
+
+        [HttpDelete]
+        [Route("")]
+        [SwaggerResponse(HttpStatusCode.OK)]
+        [SwaggerResponse(HttpStatusCode.Forbidden)]
+        [SwaggerResponse(HttpStatusCode.BadRequest)]
+        [SwaggerResponse(HttpStatusCode.NotFound)]
+        public HttpResponseMessage Delete(Guid organizationUuid, Guid unitUuid)
+        {
+            return _orgUnitService.Delete(organizationUuid, unitUuid)
+                .Match(FromOperationError, Ok);
+        }
+
+        [HttpPut]
+        [Route("registrations/{targetUnitUuid}")]
```

Comment:
why is targetUnitId in the route and not in the body content...? that does not make much sense since it is part of the change parameters, not the resource location